core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -77,13 +78,20 @@ public void build(AnalysisService analysisService) {
             tokenFilters.add(tokenFilter);
         }
 
-        int positionOffsetGap = analyzerSettings.getAsInt("position_offset_gap", 0);
-        int offsetGap = analyzerSettings.getAsInt("offset_gap", -1);
+        int positionIncrementGap = 0;
+
+        if (Version.indexCreated(indexSettings).before(Version.V_2_0_0) ){
+            positionIncrementGap = analyzerSettings.getAsInt("position_offset_gap", 0);
+        }else{
```

Comment:
I don't think that matters. This is a new setting we are adding, as far as the API is concerned. I don't think it is a problem to validate the user didn't try to configure the same thing in two different ways.